suite-native/device-manager/src/components/DeviceList.tsx

```diff
@@ -122,81 +121,69 @@ export const DeviceList = ({ isVisible, onSelectDevice }: DeviceListProps) => {
         });
     };
 
-    //to hide/show the list with animation
     useEffect(() => {
-        if (isVisible) {
-            const otherDevicesHeight = notSelectedInstancelessDevices.length * ITEM_HEIGHT;
-
-            const separatorHeight =
-                hasUnselectedDevices && !isDiscoveryActive
-                    ? 2 * SEPARATOR_VERTICAL_PADDING + SEPARATOR_HEIGHT
-                    : 0;
-
-            const buttonHeight = isDiscoveryActive ? 0 : BUTTON_PADDING_TOP + BUTTON_HEIGHT;
-
-            const h =
-                MANAGER_MODAL_BOTTOM_RADIUS + //top margin for radius
-                PADDING_TOP +
-                otherDevicesHeight + // all other device
-                separatorHeight + // separator if there is one
-                buttonHeight +
-                PADDING_BOTTOM;
-
-            opacity.value = withDelay(100, withTiming(1, { duration: 200 }));
-            height.value = withTiming(h, { duration: 300 });
-
-            setIsShown(true);
-        } else {
-            opacity.value = withTiming(0, { duration: 300 });
-            height.value = withDelay(
-                100,
-                withTiming(0, { duration: 200 }, () => {
-                    runOnJS(setIsShown)(false);
-                }),
-            );
-        }
+        'worklet';
+
+        const otherDevicesHeight = notSelectedInstancelessDevices.length * ITEM_HEIGHT;
+
+        const separatorHeight =
+            hasUnselectedDevices && !isDiscoveryActive
+                ? 2 * SEPARATOR_VERTICAL_PADDING + SEPARATOR_HEIGHT
+                : 0;
+
+        const buttonHeight = isDiscoveryActive ? 0 : BUTTON_PADDING_TOP + BUTTON_HEIGHT;
+
+        const h =
+            MANAGER_MODAL_BOTTOM_RADIUS + //top margin for radius
+            PADDING_TOP +
+            otherDevicesHeight + // all other device
+            separatorHeight + // separator if there is one
+            buttonHeight +
+            PADDING_BOTTOM -
+            MANAGER_MODAL_BOTTOM_RADIUS;
+
+        height.value = isVisible ? h : 0;
+        opacity.value = isVisible ? 1 : 0;
     }, [
+        hasUnselectedDevices,
         height,
+        isDiscoveryActive,
         isVisible,
-        opacity,
         notSelectedInstancelessDevices.length,
-        utils.spacings.small,
-        hasUnselectedDevices,
-        isDiscoveryActive,
+        opacity,
+        utils.borders.widths.small,
     ]);
 
     const listAnimatedStyle = useAnimatedStyle(
         () => ({
-            opacity: opacity.value,
-            height: height.value,
+            opacity: withTiming(opacity.value, { duration: ANIMATION_DURATION }),
```

Comment:
Please move the withTiming to the place where is the value set. So to the `useDerivedValue` or the `useEffect`, depends how you will resolve my previous comment.

Reply:
10e7175793